bdd(nfs-pvc retain policy): add integration test to verify volume-events-exporter with retain policy

mittachaitu opened the pull request:
**What this PR does**:
This PR adds integration tests to verify volume-events-exporter
behavior when NFS PVC with Reclaim policy retained is provisioned.
Volume-events-exporter should be sent events only when NFS-PV is
deleted but not the NFS PVC.

**Test case & expectation**:
| Action | Expected Behavior |
| -------- | ------------------------- |
| Create NFS PVC with Reclaim Policy Retain | Verify volume_event_controller should send `nfs_pvc`, `nfs_pv`, `backend_pvc` & `backend_pv` details to configured server |
| Delete NFS PVC with Reclaim Policy Retain | volume_event_controller should not send events to server |
| Delete NFS PV which is in Released state | Verify `volume_event_controller` should send `nfs_pv`, `backend_pvc` & `backend_pv` details to configured server(nfs_pv MUST contains creation & deletion timestamp) |


**Note to reviewers**
- This PR can be merged after #2 



Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

mynktl commented on tests/README.md:
```diff
@@ -24,5 +24,7 @@ Following parameters can be configured as a arguments to test:
+| Create NFS PVC with reclaim policy Delete | <ul> <li> Verify whether volume_event_controller sent volume_provisioned JSON data to configured server <ul> <li> nfs_pvc </li> <li> nfs_pv </li> <li> backend_pvc </li> <li> backend_pv </li> along with creationTimestamp on `nfs_pv` </ul> </ul> </li>| [nfs_sanity_test.go](./nfs_sanity_test.go) |
+| Delete NFS PVC with reclaim policy Delete | <ul> <li> Verify whether volume_event_controller sent volume_deprovisioned JSON data to configured server <ul> <li> nfs_pv </li> <li> backend_pvc </li> <li> backend_pvc </li> along with creation and deletion timestamp on `nfs_pv` </ul> </ul> </li> | [nfs_sanity_test.go](./nfs_sanity_test.go) |
+| Create NFS PVC with ReclaimPolicy Retain | <ul> <li> Verify whether volume_event_controller sent volume_provisioned JSON data to configured server <ul> <li> nfs_pvc </li> <li> nfs_pv </li> <li> backend_pvc </li> <li> backend_pv </li> along with creationTimestamp on `nfs_pv` </ul> </ul> </li> | [nfs_pvc_retain_test.go](./nfs_pvc_retain_test.go) |
+| Delete NFS PVC with ReclaimPolicy Retain | <ul> <li> volume_event_controller shouldn't send event when `nfs_pvc` is deleted </li> <li> Verify whether volume_event_controller sent volume_provisioned JSON data to configured server upon `nfs_pv` deletion <ul> <li> nfs_pv </li> <li> backend_pvc </li> <li> backend_pv </li> along with creation and deletion timestamp on `nfs_pv` </ul> </ul> </li> | [n ...
```
let's add a separate entry for `Destroy NFS PV with ReclaimPolicy Retain`

mittachaitu replied:
Updated

mynktl commented on tests/nfs_pvc_retain_test.go:
```diff
@@ -0,0 +1,273 @@
+		It("should not send events to server", func() {
+			var isDeleteEventReceived bool
+			for retry := 5; retry >= 0; retry-- {
+				backendPVCObj, err := Client.getPVC(OpenEBSNamespace, backendPVCName)
```
nit: better to verification for non-empty `backendPVCName`. also `nfsPVName` and `backendPVName` wherever required.

mittachaitu replied:
Done

Commit:
bdd(nfs-pvc retain): add integration test to verify volume-events-exporter behavior with reclaim retain policy

This PR adds integration tests to verify volume-events-exporter
behavior when NFS PVC with Reclaim policy retain is provisioned &
de-provisioned. Volume-events-exporter should be sent events during
provisioing and delete events only when NFS-PV is deleted not the NFS PVC.

Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>
